src/lib/addr.h

```diff
@@ -0,0 +1,29 @@
+#ifndef ADDR_H_
+#define ADDR_H_
+
+#include <sys/socket.h>
+
+enum {
+	DQLITE_ADDR_PARSE_UNIX = 1 << 0 /* Parse Unix socket addresses in @ notation */
+};
+
+/** Parse a socket address from the string @input.
+ *
+ * On success, the resulting address is placed in @addr, and its size is placed
+ * in @addr_len. If @addr is not large enough (based on the initial value of
+ * @addr_len) to hold the result, DQLITE_ERROR is returned.
+ *
+ * @service should be a string representing a port number, e.g. "8080".
+ *
+ * @flags customizes the behavior of the function. Currently the only flag is
+ * DQLITE_ADDR_PARSE_UNIX: when this is ORed in @flags, parseAddr will also
+ * parse Unix socket addresses in the form `@NAME`, where NAME may be empty.
+ * This creates a socket address in the (Linux-specific) "abstract namespace".
+ */
+int parseAddr(const char *input,
```

Comment:
Nit: please name this `AddrParse()`.

The convention should be CamelCase capitalized names for internally exported function, with the module name as prefix. Private functions declared inside modules as `static` should be camelCase, without capital letter.  See `vfs.h`/`vfs.c` in dqlite for examples of this convention, and the whole code base in libraft.

Unfortunately most of dqlite code base does not yet follow this convention, but it'd be nice to gradually migrate to it as we modify code.

YMMV